src/Collection/Collection.service.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { v4 as uuidv4 } from 'uuid'
+import { Collection } from './Collection.model'
+import { CollectionService } from './Collection.service'
+import { wallet } from '../../spec/utils'
+
+jest.mock('./Collection.model')
+
+describe.only('Collection service', () => {
+  describe('#isLocked', () => {
+    const collectionAttributes = {
+      id: uuidv4(),
+      name: 'Test',
+      eth_address: wallet.address,
+      salt: '',
+      contract_address: '0x2a8fd99c19271f4f04b1b7b9c4f7cf264b626edb',
+      is_published: false,
+      is_approved: false,
+      minters: [],
+      managers: [],
+      forum_link: null,
+      lock: null,
+      reviewed_at: new Date(),
+      created_at: new Date(),
+      updated_at: new Date(),
+    }
+
+    const service = new CollectionService()
+    const findOneMock = Collection.findOne as jest.Mock
+    const isPublishedMock = jest.spyOn(service, 'isPublished')
+
+    const twoDaysInMilliseconds = 2 * 24 * 60 * 60 * 1000
+
+    it('should return false if the collection does not exist', async () => {
+      findOneMock.mockImplementationOnce((collectionId: string) => {
+        if (collectionAttributes.id === collectionId) {
+          return undefined
+        }
+        throw new Error('Invalid id')
+      })
+      expect(await service.isLocked(collectionAttributes.id)).toBe(false)
+    })
```

Comment:
What do you think about using a `describe` here to describe the context and a `beforeEach` to code it?
```suggestion
describe('when the collection doesn't exist', () => {
    beforeEach(() => {
      findOneMock.mockImplementationOnce((collectionId: string) => {
        if (collectionAttributes.id === collectionId) {
          return undefined
        }
        throw new Error('Invalid id')
      })
    })
    it('should return false', async () => {
      expect(await service.isLocked(collectionAttributes.id)).toBe(false)
    })
})
```

Reply:
I like it! I'm not really used to use describe to explain the context in these situations but I like how it looks/structures the code 👍